src/foam/nanos/boot/Boot.java

```diff
@@ -75,6 +78,18 @@ public void put(Object obj, Detachable sub) {
       }
     }, null);
 
+    // PM factory
+    root_ = root_.putFactory("PM", new XFactory() {
+      public Object create(X x) {
+        int rand = ThreadLocalRandom.current().nextInt(0, 100);
+        if ( rand == 0 ) {
+          return new PM();
+        } else {
+          return new NullPM();
```

Comment:
Don't allocate a NullPM instead use a static single instance all the time, that way we save on object allocations too.

Reply:
Changed to single instance.